modules/HapticGlove_module/src/GloveControlHelper.cpp

```diff
@@ -0,0 +1,335 @@
+    }
+    if (m_humanFingerNameList.size() != m_numFingers)
+    {
+        yError() << m_logPrefix
+                 << "the number of hand fingers in the configuration "
+                    "file is not equal to the const number of fingers";
+        return false;
+    }
+    m_jointRangeMin.resize(m_humanJointNameList.size(), 0.0);
+    m_jointRangeMax.resize(m_humanJointNameList.size(), 0.0);
+
+    // wearable device
+    m_pImp = std::make_unique<GloveWearableImpl>(
+        m_numFingers, m_numForceFeedback, m_numVibrotactileFeedback, m_numHandJoints);
+    if (!m_pImp->configure(config, name, m_isRightHand))
+    {
+        yError() << m_logPrefix << "unable to configure the haptic glove wearable device.";
+        return false;
+    }
+
+    if (!this->setupGlove())
+    {
+        yError() << m_logPrefix << "cannot setup the glove.";
+        return false;
+    }
+    yInfo() << m_logPrefix << "configuration is done.";
+
+    return true;
+}
+
+bool GloveControlHelper::getFingertipPoses(Eigen::MatrixXd& measuredValues)
+{
+    return m_pImp->getFingertipPoseValues(measuredValues);
+}
+
+bool GloveControlHelper::getHandJointAngles(std::vector<double>& jointAngles)
+{
+    return m_pImp->getJointValues(jointAngles);
+}
+
+bool GloveControlHelper::setFingertipForceFeedbackReferences(
+    const std::vector<double>& desiredValue)
+{
+    if (desiredValue.size() != m_numForceFeedback)
+    {
+        yError() << m_logPrefix
+                 << "the size of the input "
+                    "desired vecotr ["
+                 << desiredValue.size() << "] and the number of haptic force feedbacks [ "
+                 << m_numForceFeedback << " ]are not equal.";
+        return false;
+    }
+
+    for (size_t i = 0; i < m_numForceFeedback; i++)
+    {
+        m_desiredForceValues[i]
+            = (int)std::round(std::max(0.0, std::min(desiredValue[i], m_maxForceFeedback)) * 100
+                              / m_maxForceFeedback);
+    }
+
+    return m_pImp->setFingertipForceFeedbackValues(m_desiredForceValues);
+}
+
+bool GloveControlHelper::setFingertipVibrotactileFeedbackReferences(
+    const std::vector<double>& desiredValue)
+{
+
+    if (desiredValue.size() != m_numVibrotactileFeedback)
+    {
+        yError() << m_logPrefix << "the size of the input desired vecotr [" << desiredValue.size()
+                 << "] and the number of haptic force feedbacks [ " << m_numVibrotactileFeedback
+                 << " ]are not equal.";
+        return false;
+    }
+
+    for (size_t i = 0; i < m_numVibrotactileFeedback; i++)
+    {
+        m_desiredVibrotactileValues[i]
+            = (int)std::round(std::max(0.0, std::min(desiredValue[i], 100.0)));
+    }
+
+    return m_pImp->setFingertipVibrotactileValues(m_desiredVibrotactileValues);
+}
+bool GloveControlHelper::stopPalmVibrotactileFeedback()
+{
+    return m_pImp->setPalmVibrotactileValue(
+        124); // to turn off set `124`, for more details check the documentation.
```

Comment:
here is the different types of commands we can set to the glove plam buzz motor: https://github.com/robotology/wearables/blob/ff6bd84a98f3fc01dc44e09af24475c2c72ce1ae/devices/HapticGlove/include/SenseGloveHelper.hpp#L38-L70
As you can see, the order of those values have not specific meaning. Moreover, through wearable we can set a value to wearable port. Here a guide is provided what are the permissible values.

Reply:
My suggestion was to add a method in https://github.com/robotology/wearables/blob/ff6bd84a98f3fc01dc44e09af24475c2c72ce1ae/devices/HapticGlove/include/SenseGloveHelper.hpp to disable the palm feedback. In this way, if you want to use other gloves that use different numbers, you don't have to change this line. Do you see what I mean?

Reply:
Yes, I know what you mean and it sounds reasonable. The problem is related to wearable interface which does not allow to have such interface. Through Wearables, we can only set a value for a haptic type name and stream it to the dedicated port.
What instead we can do, to avoid having that magic number (which has already the explanation), I copy https://github.com/robotology/wearables/blob/ff6bd84a98f3fc01dc44e09af24475c2c72ce1ae/devices/HapticGlove/include/SenseGloveHelper.hpp#L38-L70 to walking-teleoperation project, and set by enum value and doing the casting.
Let me know @S-Dafarra  if is sounds like a good compromise?

Reply:
I am still not convinced. That enum contains a series of magic numbers and tags (e.g. ``Cue_Game_Over``) that make sense only for the senseglove.

I would have expected the driver to reinterpret a generic input signal (for example 0 -> no feedback, 124 -> max feedback, or a more straightforward set of tags) into something that the senseglove would understand. 

In any case, from what I understood, in order to clean this part, we would also need another PR on the wearable side. Since this is already complex on its own, I would suggest spawning a specific issue on this and move on.

Reply:
> I guess you can also remove that line. By default it should be `off`

It is not. Look at here: 

https://github.com/robotology/yarp/blob/165a46c6e034fc27b90cc0bd80b35419679b22c3/src/devices/RemoteControlBoard/RemoteControlBoard.h#L96-L97

Reply:
With this commit https://github.com/robotology/walking-teleoperation/pull/72/commits/8d336ef917ebb993c42b0bcd05db10d9d30db316, I have added the enum class for the senseGlove thumper commands. And not the stop commands use the enum value to send stopping value.

Reply:
related issue https://github.com/robotology/walking-teleoperation/issues/86

Closing this conversation. It can followed in the issue.